assignment1/src/main/java/ro/utcn/alecostantea/assignment1/controller/QuestionController.java

```diff
@@ -0,0 +1,41 @@
+package ro.utcn.alecostantea.assignment1.controller;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.*;
+import ro.utcn.alecostantea.assignment1.command.Command;
+import ro.utcn.alecostantea.assignment1.dto.QuestionDTO;
+import ro.utcn.alecostantea.assignment1.model.Question;
+import ro.utcn.alecostantea.assignment1.service.QuestionService;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@RestController
+@RequiredArgsConstructor
+public class QuestionController {
+    private final QuestionService questionService;
+    private List<Command> commandList = new ArrayList<>();
+
+    @GetMapping("/questions")
+    public List<Question> readAll() {
+        return questionService.findAll();
+    }
+
+    @PostMapping("/questions")
+    public void addQuestion(@RequestBody QuestionDTO dto){
+        //System.out.println("java" + dto);
+
+        Command command = questionService.addQuestion(dto);
+        commandList.add(command);
+    }
+    @GetMapping("/questions/title/{title}")
```

Comment:
For a better REST API, I would suggest using `@RequestParam`s instead of `@PathVariables` for the filtering stuff. In the end, for a REST perspective, all three query endpoints (findAll, findByTag and findByTitle) are just one big query endpoint with different filter criteria.